option.go

```diff
@@ -5,14 +5,27 @@ import (
 	"fmt"
 	"reflect"
 	"regexp"
+
+	"emperror.dev/errors"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	defaultSource  = "default"
+	overrideSource = "override"
+	promptSource   = "prompt"
+	yamlSource     = "yaml"
+	jsonSource     = "json"
 )
 
 type option interface {
 	IsDefined() bool
 	GetValue() any
 	SetValue(any) error
-	SetSource(string)
-	GetSource() string
+	SetSource(SourceLocation)
+	GetSource() SourceLocation
```

Comment:
I generally agree, but we cant use just use `Source()` because `Source` is the main property in the Option:
```
type Option[T any] struct {
	Source  SourceLocation
	Defined bool
	Value   T
}
```
Not sure if you have a suggestion for this ...

Reply:
Ah yeah, I've run into this many times.